.github/workflows/windows-pull-request.yaml

```diff
@@ -0,0 +1,78 @@
+name: Windows tests (Pull Request)
+
+on:
+  pull_request:
+    branches:
+      - main
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - labeled
+      - unlabeled
+
+permissions: {}
+
+# Limit concurrency by workflow/branch combination.
+#
+# For pull request builds, pushing additional changes to the
+# branch will cancel prior in-progress and pending builds.
+#
+# For builds triggered on a branch push, additional changes
+# will wait for prior builds to complete before starting.
+#
+# https://docs.github.com/en/actions/using-jobs/using-concurrency
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
+jobs:
+  run-tests-sqlite:
+    name: Test with SQLite
+    if: contains(github.event.pull_request.labels.*.name, 'arch:windows')
+
+    strategy:
+      matrix:
+        python-version:
+          - "3.9"
+          - "3.10"
+
+      fail-fast: false
+
+    permissions:
+      contents: read
+
+    runs-on: windows-latest
+    timeout-minutes: 45
+
+    env:
+      # enable colored output
+      # https://github.com/pytest-dev/pytest/issues/7443
+      PY_COLORS: 1
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+
+      - name: Set up Docker Buildx
+        if: ${{ matrix.build-docker-images }}
+        uses: docker/setup-buildx-action@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: "pip"
+          cache-dependency-path: "requirements*.txt"
+
```

Comment:
Should we actually be using the `docker/build-push-action`?

Reply:
Or do we actually not need either docker steo?

Reply:
Ah, you're right! We don't need Docker, because anything running in Docker on Windows would be running in a Linux VM, which kinda defeats the purpose of these tests 🙃 That is, unless we're using Windows containers, but those never really got much adoption, as far as I know

Anyways, I've removed the code that sets up buildx